lib/Parse/ParseExpr.cpp

```diff
@@ -2621,8 +2622,11 @@ ParserStatus Parser::parseClosureSignatureIfPresent(
         // It is a common error to try to capture a nested field instead of just
         // a local name, reject it with a specific error message.
         if (Tok.isAny(tok::period, tok::exclaim_postfix,tok::question_postfix)){
-          diagnose(Tok, diag::cannot_capture_fields);
-          skipUntil(tok::comma, tok::r_square);
+          auto diag = diagnose(Tok, diag::cannot_capture_fields);
+          skipTo(tok::comma, tok::r_square);
+          StringRef name = Tok.getText();
```

Comment:
You should limit this to `tok::identifier` token. For example  `{ [foo.bar(baz), ... ] in` in this case, we should not predicate the desired name. Either giving up emitting fix-it, or emit like `<#name#> = ` (in `DiagnosticsEditorMode`)
Also, it'd be great if we can escape it when the token is a keyword.

Reply:
done